src/reason_pprint_ast.ml

```diff
@@ -3089,7 +3096,12 @@ class printer  ()= object(self:'self)
           | Ppat_lazy p ->formatPrecedence (label ~space:true (atom "lazy") (self#simple_pattern p))
           | Ppat_extension e -> self#extension e
           | Ppat_exception p ->
-              makeList ~postSpace:true [atom "exception"; self#pattern p]
+              (*
+                An exception pattern should be wrapped in (...)
```

Comment:
Also, at *parse* time we are a little more liberal - we accept things like:

```
| exception IntAndString i s =>...
```

It's just that the rules for what goes to the right of the exception are a little (too) nuanced. It accepts "non simple" parameters, *except* in the case of `as`. So with your diff, it consistently applies "simplification" to the `exception` argument, and that seems fine to me (I think as I mentioned we should use the `simple_pattern` method though).